core/dbt/events/functions.py

```diff
@@ -108,16 +109,27 @@ def scrub_secrets(msg: str, secrets: List[str]) -> str:
     return scrubbed
 
 
-# Type representing Event and all subclasses of Event
-T_Event = TypeVar('T_Event', bound=Event)
+# returns a dictionary representation of the event fields. You must specify which of the
+# available messages you would like to use (i.e. - e.message, e.cli_msg(), e.file_msg())
+# used for constructing json formatted events. includes secrets which must be scrubbed at
+# the usage site.
+def event_to_dict(e: T_Event, msg_fn: Callable[[T_Event], str]) -> dict:
+    level = e.level_tag()
+    return {
+        'log_version': e.log_version,
+        'ts': e.get_ts(),
+        'pid': e.get_pid(),
+        'msg': msg_fn(e),
+        'level': level if len(level) == 5 else f"{level} "
```

Comment:
Since this is for the json output we don't want to format the level here.  Just in the text.